features/323-login-id/design.md

```diff
@@ -0,0 +1,165 @@
+**NOTE: this specification deprecates #294: Login without specifiying login ID key**
+
+# Background
+
+## Current Design
+
+Under current design of authentication:
+
+1. Each **User** is associated with some **Principal**.
+3. Each **Principal** is associated with one **Identity Provider** (e.g. password authentication, OAuth).
+4. Password authentication principals is represented by tuple (**Login ID Key**, **Login ID**, password hash).
+5. App developer specifies a list of white-listed login ID key.
```

Comment:
In existing design, the Login ID Key is not really whitelisted -- default it allows any login key to avoid user confusion, and only whitelisted if user give it a setting.

IF we make login ID have semantic meaning -- i.e. most of the time people are expected to only use`email`, `phone`, `username` as login ID key, I think we can change to really whitelisted by default.